oe_theme.theme

```diff
@@ -418,6 +420,27 @@ function oe_theme_preprocess_pattern_dialog(array &$variables): void {
   $variables['hidden'] = $variables['hidden'] ? 'true' : 'false';
 }
 
+/**
+ * Implements hook_preprocess_HOOK() for pattern gallery template.
+ */
+function oe_theme_preprocess_pattern_gallery(array &$variables): void {
+  $variables['rows'] = [];
+
+  $count_item = 0;
+  foreach ($variables['items'] as $item) {
+    $item += ['classes' => 'ecl-col-md-3'];
+
+    // Format the icon for ecl.
+    if (!empty($item['icon'])) {
+      $item['icon'] = 'ecl-icon--' . $item['icon'];
+    }
+
+    $item['image'] = ImageValueObject::fromArray($item['image']);
+    $variables['rows'][ceil($count_item / 4)][] = GalleryItemValueObject::fromArray($item);
```

Comment:
On a second thought splitting it into rows should be also moved in the Twig file and, most importantly, the `::fromArray()` factories should be used in  `function oe_theme_preprocess_pattern_gallery__preview()`, meaning only in the preview theme suggestion.

We did a similar thing in `pattern-file-translation.html.twig`, it's important because we have to clearly define our API here, meaning: our pattern expects a simple list of gallery item objects, all the rest will happen in the Twig template. This will guarantee that developers will be able to confidently use our patterns and that, if we need to change presentation logic, that will only happen in the Twig template.

Reply:
:+1: